pkg/controller/controller_instance.go

```diff
@@ -244,6 +254,23 @@ func (c *controller) reconcileServiceInstance(instance *v1beta1.ServiceInstance)
 	}
 }
 
+// initObservedGeneration implements ObservedGeneration initialization based on
+// ReconciledGeneration for status API migration
+func (c *controller) initObservedGeneration(instance *v1beta1.ServiceInstance) (bool, error) {
```

Comment:
Can it be in the conversions? I had the same thought, but figured that the conversions only worked for converted between discrete versions, say v1beta1 to v1. Or is converting to the internal version a place where the conversions could add `ObservedGeneration`? If so, would it be strange that the added `ObservedGeneration` is only written to storage if the controller happens to make another change?

Reply:
I don't think it can be in the conversions for 2 reasons:
1. We don't declare a new API version here
2. I want to update the etcd data here to be able to delete `ReconciledGeneration` from API completely once we're done with the migration.

Reply:
> would it be strange that the added ObservedGeneration is only written to storage if the controller happens to make another change?

@staebler not sure if I understood you correctly, but all Service Catalog resources will be re-processed after restart of `controller-manager`, which will definitely happen as part of updating a svc-cat Docker image to the newer version in the Pod/Deployment.

Reply:
> >    would it be strange that the added ObservedGeneration is only written to storage if the controller happens to make another change?

> @staebler not sure if I understood you correctly, but all Service Catalog resources will be re-processed after restart of controller-manager, which will definitely happen as part of updating a svc-cat Docker image to the newer version in the Pod/Deployment.

@nilebox I was envisioning that the conversion would happen between the external v1beta1 version and the internal version. The conversion functions would not necessitate a new API version. When the API server reads in from storage an instance, the conversion function would add values for `ObservedGeneration` and `ProvisionStatus` if needed. However, those values would only make it back to etcd upon an update of the instance. Consequently, if there is a service instance from before this change that exists, is provisioned, and is never updated, then it will never have values for `ObservedGeneration` and `ProvisionStatus` in storage. The user would still see values for those fields, though, since they would be requesting the instance via the API server.

Reply:
> However, those values would only make it back to etcd upon an update of the instance. Consequently, if there is a service instance from before this change that exists, is provisioned, and is never updated, then it will never have values for ObservedGeneration and ProvisionStatus in storage. 

I agree, and that's why I would prefer to keep it as it is (i.e. no API conversion). With the current code `ObservedGeneration` and `ProvisionStatus` will be initialized as soon as you update the Service Catalog Docker image to this version (and consequently restart the controller-manager pod).
This would allow to remove `ReconciledGeneration` as soon as we think that there are no users  still using versions of Service Catalog prior to introduction of `ObservedGeneration` and `ProvisioningStatus`. Does it make sense?